lib/signex/helper.ex

```diff
@@ -40,4 +27,22 @@ defmodule SignEx.Helper do
   defp hash_key(key) do
     :crypto.hash(:md5, key)
   end
+
+  def fetch_keys(collection, keys) do
+    collection = for {k, v} <- collection, do: {k, v}
+    do_fetch_keys(collection, keys, [])
+  end
+
+  def do_fetch_keys(_collection, [], progress) do
+    {:ok, Enum.reverse(progress)}
+  end
+  def do_fetch_keys(collection, [key | rest], progress) do
+    key = "#{key}"
```

Comment:
Depends on what we expect. could just add a clause that says only accept keys that are binary.